packages/gestalt/lib/classnameBuilder.js

```diff
@@ -0,0 +1,84 @@
+// ClassnameBuilder is a hack used to help us minify Gestalt CSS classnames down
+// to three characters each for prod builds.
+//
+// The issue with such an aggressive minification approach is that it leaves a
+// very small hash value size and collisions often occur. Gestalt previously
+// solved this by assigning each classname an incrementing number and hashing on
+// that number, but this resulted in lots of name/hash changes whenever
+// classnames were added or removed. To solve this, we now hash based on the
+// class name and only add variation when hashes collide.
+
+import md5 from 'blueimp-md5';
+
+class ClassnameBuilder {
+  constructor() {
+    // Map of classnames to their minified versions.
```

Comment:
No change needed but curious if you considered using a `Map` instead - im interested if there was an internal discussion and the trade-offs https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map

Reply:
No reason. Just old habits. I'm pretty sure the way I'm using the objects as a Map makes literally 0 difference in speed, but you're right that actually making it a Map reads more explicitly.